### ISSUE TITLE
load_stac raises ValueError when STAC file uses "Z" as TZ for UTC timestamps

### ISSUE THREAD

**Issue by JohanKJSchreurs** [Open-EO/openeo-geopyspark-driver#568]
## load_stac raises ValueError when STAC file uses "Z" as TZ for UTC timestamps

`connection.load_stac(...)` raises a ValueError when the UTC timestamps in the STAC file use the TZ format with "Z", for example: '2020-01-01T00:00:00Z'
This format is commonly used, so it is best that we support it too.

This happens with Python 3.8 because `datetime.datetime.fromisoformat` did not yet support this format yet in Python 3.8.
Though, it is supported starting from Python 3.11.
See also: https://docs.python.org/3/library/datetime.html?highlight=datetime%20fromisoformat#datetime.datetime.fromisoformat

However, we can catch this inside the load_stac implementation and handle it there. 

### Exception info

Error on the client side:
openeo.rest.OpenEoApiError: [500] Internal: Server error: ValueError("Invalid isoformat string: '2020-01-01T00:00:00Z'") 

Error traceback from server logs:
```
"Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/users/auth.py", line 88, in decorated
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/views.py", line 631, in result
    result = backend_implementation.processing.evaluate(process_graph=process_graph, env=env)
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/ProcessGraphDeserializer.py", line 300, in evaluate
    return evaluate(process_graph=process_graph, env=env)
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/ProcessGraphDeserializer.py", line 370, in evaluate
    result = convert_node(result_node, env=env)
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/ProcessGraphDeserializer.py", line 390, in convert_node
    process_result = apply_process(process_id=process_id, args=processGraph.get('arguments', {}),
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/ProcessGraphDeserializer.py", line 1490, in apply_process
    args = {name: convert_node(expr, env=env) for (name, expr) in sorted(args.items())}
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/ProcessGraphDeserializer.py", line 1490, in <dictcomp>
    args = {name: convert_node(expr, env=env) for (name, expr) in sorted(args.items())}
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/ProcessGraphDeserializer.py", line 402, in convert_node
    return convert_node(processGraph['node'], env=env)
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/ProcessGraphDeserializer.py", line 390, in convert_node
    process_result = apply_process(process_id=process_id, args=processGraph.get('arguments', {}),
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/ProcessGraphDeserializer.py", line 1522, in apply_process
    return process_function(args=ProcessArgs(args, process_id=process_id), env=env)
  File "/usr/local/lib/python3.8/site-packages/openeo_driver/ProcessGraphDeserializer.py", line 2123, in load_stac
    return env.backend_implementation.load_stac(url=url, load_params=load_params, env=env)
  File "/src/openeogeotrellis/backend.py", line 1064, in load_stac
    for itm in intersecting_items:
  File "/src/openeogeotrellis/backend.py", line 1052, in <genexpr>
    for itm in intersecting_catalog.get_items() if intersects_spatiotemporally(itm))
  File "/src/openeogeotrellis/backend.py", line 877, in intersects_spatiotemporally
    return intersects_temporally() and intersects_spatially()
  File "/src/openeogeotrellis/backend.py", line 865, in intersects_temporally
    nominal_date = itm.datetime or dt.datetime.fromisoformat(itm.properties["start_datetime"])
ValueError: Invalid isoformat string: '2020-01-01T00:00:00Z'", "req_id": "r-20231106xyzetc", "user_id": "testuser"}
```

**Comment by bossie** [comment]
For reference:

re: [`datetime`](https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#properties-object) (and `start_datetime`/`end_datetime`):
> It is formatted according to [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6)

which allows the Z:
> time-offset     = "Z" / time-numoffset

**Comment by bossie** [comment]
Fixed and added a test in #673.

**Comment by JohanKJSchreurs** [comment]
> Fixed and added a test in #673.

Thanks  @bossie !
